augur/io/sqlite3.py

```diff
@@ -0,0 +1,79 @@
+import os
+import pandas as pd
+import sqlite3
+
+
+class Sqlite3Database:
+    """Represents a SQLite3 database.
+
+    Provides useful methods to manage connections and general I/O.
+    """
+
+    def __init__(self, file:str):
+
+        self.file = file
+        """Database file."""
+
+    def connect(self, **connect_kwargs) -> sqlite3.Connection:
+        """Return a new connection to the SQLite database."""
+        return sqlite3.connect(self.file, **connect_kwargs)
+
+    def execute(self, *args):
+        """Execute under a connection context."""
+        with self.connect() as con:
+            con.execute(*args)
+
+    def executemany(self, *args):
+        """Executemany under a connection context."""
+        # TODO: benchmark performance with cursor.executemany / con.commit at the end
+        with self.connect() as con:
+            con.executemany(*args)
```

Comment:
> I think we'd want to hold the connection open across several execute statements, rather than connecting each time.

This is a good point. I'm thinking of making the `Sqlite3Database` a context manager so it can be used like this, where metadata and sequences each get their own connection:

```py
with Sqlite3Database(db_file) as database:
    database.create_table("metadata")
    database.executemany(insert_statement, metadata.rows)
    database.create_unique_index("metadata", "strain")
with Sqlite3Database(db_file) as database:
    database.create_table("sequences")
    database.executemany(insert_statement, sequences.records)
    database.create_unique_index("sequences", "strain")
```

Reply:
> But stepping back, why make these wrappers for `execute()` and `executemany()` in the first place?

I forget why I made it like this. There was probably a good reason in the other PR, but not for here.

However, with the context manager above, these wrappers could do more, e.g.

```py
def execute(self, *args):
    if self.connection:
        self.connection.execute(*args)
    else:
        with self.connect() as connection:
            connection.execute(*args)
```

Reply:
Implemented in b3c93f3a810d2558b0bcd1a878531f1b315c6725...aba56fe1c4811aa8fbde59e129776138af646c80.